File: xide-ui/src/main/java/org/xast/xide/ui/components/code_panel/neo_editor/NeoEditorView.java
package org.xast.xide.ui.components.code_panel.neo_editor;

import java.awt.BorderLayout;
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

import org.xast.xide.core.event.EventBus;
import org.xast.xide.core.event.FileSaveRequestedEvent;
import org.xast.xide.core.plugin.file.FileModel;
import org.xast.xide.core.plugin.file.TextFileModel;
import org.xast.xide.core.plugin.ui.CodePanelView;
import org.xast.xide.core.utils.Debug;

public class NeoEditorView extends CodePanelView {
    private NeoEditor neoEditor;
    private NeoEditorStatus editorStatus;

    public NeoEditorView(EventBus eventBus, File file) {
        setLayout(new BorderLayout());

        String content = new String();

        if (file.exists() && file.canRead()) {
            try {
                content = Files.readString(file.toPath());
            } catch (IOException e) { 
                Debug.error("Cannot read file `"+file.getName()+"`: "+e.getMessage());
            }
        }

        editorStatus = new NeoEditorStatus();
        add(editorStatus, BorderLayout.SOUTH);

        neoEditor = new NeoEditor(
            content, 
            editorStatus, 
            () -> {
                eventBus.publish(new FileSaveRequestedEvent(file, false));
            }
        );
        add(neoEditor, BorderLayout.CENTER);
    }

    @Override
    public FileModel model() {
        return new TextFileModel(neoEditor.getContent());
    }
}

File: xide-ui/src/main/scala/org/xast/xide/ui/components/code_panel/neo_editor/NeoEditorView.scala
package org.xast.xide.ui.components.code_panel.neo_editor

import java.awt.BorderLayout
import java.io.File
import java.io.IOException
import java.nio.file.Files

import org.xast.xide.core.event.EventBus
import org.xast.xide.core.event.FileSaveRequestedEvent
import org.xast.xide.core.plugin.file.FileModel
import org.xast.xide.core.plugin.file.TextFileModel
import org.xast.xide.core.plugin.ui.CodePanelView
import org.xast.xide.core.utils.Debug

class NeoEditorView(val eventBus: EventBus, val file: File) extends CodePanelView:
    private var neoEditor: NeoEditor = scala.compiletime.uninitialized
    private var editorStatus: NeoEditorStatus = scala.compiletime.uninitialized

    setLayout(new BorderLayout())

    private var content = ""

    if (file.exists() && file.canRead()) {
        try {
            content = Files.readString(file.toPath())
        } catch {
            case e: IOException =>
                Debug.error("Cannot read file `" + file.getName() + "`: " + e.getMessage())
        }
    }

    editorStatus = new NeoEditorStatus()
    add(editorStatus, BorderLayout.SOUTH)

    neoEditor = new NeoEditor(
        eventBus,
        content,
        editorStatus,
        () => eventBus.publish(new FileSaveRequestedEvent(file, false)),
    )
    add(neoEditor, BorderLayout.CENTER)

    override def model(): FileModel = new TextFileModel(neoEditor.getContent())
